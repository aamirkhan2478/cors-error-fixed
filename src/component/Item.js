/* eslint-disable react/prop-types */
import { AiOutlineRight } from 'react-icons/ai';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { getdetail } from '../redux/popup/popup';

const Item = ({ record, count }) => {
  const dispatch = useDispatch();
  const handleClick = (record) => {
    dispatch(getdetail(record));
  };

  const image = `${process.env.PUBLIC_URL}/images/${record.name}.jpg`;
  return (
    <div className={count % 2 ? 'cards right' : 'cards left'}>
      <img src={image} alt={record.name} />
      <div>
        <p>{record.name}</p>
        <button onClick={() => handleClick(record)} type="button">
          <Link to="/details"><AiOutlineRight /></Link>
        </button>
      </div>
    </div>
  );
};

export default Item;
