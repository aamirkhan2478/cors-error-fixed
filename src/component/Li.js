/* eslint-disable react/prop-types */
import React from 'react';

const Li = ({ nutritions }) => (
  <div className="li">
    <ul>
      <li>
        <p> calories: </p>
        <p>
          {' '}
          {nutritions.calories}
        </p>
      </li>
    </ul>
    <li>
      <p> carbohydrates: </p>
      <p>
        {' '}
        {nutritions.carbohydrates}
        {' '}
      </p>
    </li>
    <li>
      <p> fat: </p>
      <p>
        {' '}
        {nutritions.fat}
        {' '}
      </p>
    </li>
    <li>
      <p> protein: </p>
      <p>
        {' '}
        {nutritions.protein}
        {' '}
      </p>
    </li>
    <li>
      <p>   sugar: </p>
      <p>
        {' '}
        {nutritions.sugar}
        {' '}
      </p>
    </li>
  </div>
);

export default Li;
