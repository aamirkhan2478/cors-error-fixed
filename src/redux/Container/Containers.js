/* eslint no-param-reassign: "error" */
import { createSlice } from '@reduxjs/toolkit';

const STATUSES = Object.freeze({
  IDLE: 'idle',
  ERROR: 'error',
  LOADING: 'loading',
  SUCCESS: 'success',
});

export const ContainerSlice = createSlice({
  name: 'container',
  initialState: {
    data: [],
    status: STATUSES.IDLE,
  },
  reducers: {
    getdata: (state, action) => { state.data = action.payload; },
    setStatus: (state, action) => { state.status = action.payload; },
  },
});

export const { getdata, setStatus } = ContainerSlice.actions;

export default ContainerSlice.reducer;

// Thunks

export function fetchdata() {
  return async function fetchdataThunk(dispatch) {
    dispatch(setStatus(STATUSES.LOADING));
    try {
      const res = await fetch('https://proxy.cors.sh/https://www.fruityvice.com/api/fruit/all');
      const data = await res.json();

      dispatch(getdata(data));
      dispatch(setStatus(STATUSES.SUCCESS));
    } catch (error) {
      dispatch(setStatus(STATUSES.ERROR));
    }
  };
}
