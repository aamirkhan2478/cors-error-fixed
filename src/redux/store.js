import { configureStore } from '@reduxjs/toolkit';
import ContainerSliceReducer from './Container/Containers';
import popupSliceReducer from './popup/popup';
import searchSliceReducer from './Search/search';

export default configureStore({
  reducer: {
    popup: popupSliceReducer,
    container: ContainerSliceReducer,
    search: searchSliceReducer,
  },
});
