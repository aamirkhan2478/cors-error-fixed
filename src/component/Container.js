import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { AiOutlineArrowUp, AiFillCloseCircle, AiOutlineSearch } from 'react-icons/ai';
import { fetchdata } from '../redux/Container/Containers';
import Item from './Item';

const Container = () => {
  const { data } = useSelector((state) => state.container);
  const [scroll, setscroll] = useState(document.documentElement.scrollTop);
  const [bool, setbool] = useState(false);
  const dispatch = useDispatch();

  // const checkData = async () => {
  //   const res = await fetch('https://proxy.cors.sh/https://www.fruityvice.com/api/fruit/all');
  //   const data = await res.json();
  //   console.log(data);
  // }

  useEffect(() => {
    dispatch(fetchdata());
    window.onscroll = () => setscroll(document.documentElement.scrollTop);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  console.log(data);
  const handleScroll = () => {
    document.body.scrollTop = 0;
    document.documentElement.scrollTop = 0;
    setscroll(0);
  };

  const handleSearch = (event) => {
    const title = document.querySelector('.inputText').value;
    const cat = document.querySelector('#category').value;
    if (event.key === 'Enter') {
      dispatch(fetchdata(`${cat}/${title}`));
      document.querySelector('.sp').innerHTML = 'Searched Fruits';
    }
  };

  let count = 0;
  return (
    <div className="container">
      <div className="search">
        {bool ? (
          <>
            <div>
              <select name="category" id="category">
                <option default> select</option>
                <option value="family">Family</option>
                <option value="genus">Genus</option>
                <option value="order">Order</option>
              </select>
              <input type="text" placeholder="Enter family/Genus/Order" className="inputText" onKeyDown={() => handleSearch(event)} />
              <button
                type="button"
                onClick={() => { dispatch(fetchdata('all')); document.querySelector('.inputText').value = ''; setbool(!bool); }}
              >
                <AiFillCloseCircle />
              </button>
            </div>
            {bool ? <p className="sp" /> : ''}
          </>
        )
          : <button type="button" className="sbtn" onClick={() => setbool(!bool)}><AiOutlineSearch /></button>}
      </div>

      {data.map((record) => {
        count += 1;
        return <Item record={record} key={record.id} count={count} />;
      })}
      <div className="topbtn">
        <button className={scroll > 20 ? 'scrollbtn ' : 'scrollDown'} type="button" onClick={() => handleScroll()}><AiOutlineArrowUp /></button>
      </div>
    </div>
  );
};

export default Container;
