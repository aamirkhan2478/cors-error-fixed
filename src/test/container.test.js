import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Item from '../component/Item';
import Li from '../component/Li';
import Container from '../component/Container';

const fruit = {
  genus: 'Fragaria',
  name: 'Strawberry',
  id: 3,
  family: 'Rosaceae',
  order: 'Rosales',
  nutritions: {
    carbohydrates: 5.5,
    protein: 0,
    fat: 0.4,
    calories: 29,
    sugar: 5.4,
  },
};

test('Demo test', () => {
  expect((1 + 2)).toBe(3);
});

test('renders container Component', () => {
  const con = render(
    <BrowserRouter>
      <Provider store={store}>
        <Container />
      </Provider>
    </BrowserRouter>,
  );
  expect(con).toMatchSnapshot();
});

test('renders Item Component', () => {
  const item = render(
    <BrowserRouter>
      <Provider store={store}>
        <Item record={fruit} key={fruit.id} count={0} />
      </Provider>
    </BrowserRouter>
    ,
  );
  expect(item).toMatchSnapshot();
});

test('renders LI Component', () => {
  const li = render(
    <Provider store={store}>
      <Li nutritions={fruit.nutritions} />
    </Provider>,
  );
  expect(li).toMatchSnapshot();
});
