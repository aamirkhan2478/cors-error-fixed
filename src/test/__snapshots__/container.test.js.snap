// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Item Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cards left"
      >
        <img
          alt="Strawberry"
          src="/images/Strawberry.jpg"
        />
        <div>
          <p>
            Strawberry
          </p>
          <button
            type="button"
          >
            <a
              href="/details"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                />
              </svg>
            </a>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cards left"
    >
      <img
        alt="Strawberry"
        src="/images/Strawberry.jpg"
      />
      <div>
        <p>
          Strawberry
        </p>
        <button
          type="button"
        >
          <a
            href="/details"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
              />
            </svg>
          </a>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders LI Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="li"
      >
        <ul>
          <li>
            <p>
               calories: 
            </p>
            <p>
               
              29
            </p>
          </li>
        </ul>
        <li>
          <p>
             carbohydrates: 
          </p>
          <p>
             
            5.5
             
          </p>
        </li>
        <li>
          <p>
             fat: 
          </p>
          <p>
             
            0.4
             
          </p>
        </li>
        <li>
          <p>
             protein: 
          </p>
          <p>
             
            0
             
          </p>
        </li>
        <li>
          <p>
               sugar: 
          </p>
          <p>
             
            5.4
             
          </p>
        </li>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="li"
    >
      <ul>
        <li>
          <p>
             calories: 
          </p>
          <p>
             
            29
          </p>
        </li>
      </ul>
      <li>
        <p>
           carbohydrates: 
        </p>
        <p>
           
          5.5
           
        </p>
      </li>
      <li>
        <p>
           fat: 
        </p>
        <p>
           
          0.4
           
        </p>
      </li>
      <li>
        <p>
           protein: 
        </p>
        <p>
           
          0
           
        </p>
      </li>
      <li>
        <p>
             sugar: 
        </p>
        <p>
           
          5.4
           
        </p>
      </li>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders container Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="search"
        >
          <button
            class="sbtn"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
              />
            </svg>
          </button>
        </div>
        <div
          class="topbtn"
        >
          <button
            class="scrollDown"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M868 545.5L536.1 163a31.96 31.96 0 0 0-48.3 0L156 545.5a7.97 7.97 0 0 0 6 13.2h81c4.6 0 9-2 12.1-5.5L474 300.9V864c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V300.9l218.9 252.3c3 3.5 7.4 5.5 12.1 5.5h81c6.8 0 10.5-8 6-13.2z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="search"
      >
        <button
          class="sbtn"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
            />
          </svg>
        </button>
      </div>
      <div
        class="topbtn"
      >
        <button
          class="scrollDown"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M868 545.5L536.1 163a31.96 31.96 0 0 0-48.3 0L156 545.5a7.97 7.97 0 0 0 6 13.2h81c4.6 0 9-2 12.1-5.5L474 300.9V864c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V300.9l218.9 252.3c3 3.5 7.4 5.5 12.1 5.5h81c6.8 0 10.5-8 6-13.2z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
